Update service blocks to use a node route context

andybroomfield opened the pull request:
Fix #116

Changes the way services blocks get the current node, by changing from using the current route in the constructor and using a context mapping for the node route.
This allows the correct contents of the blocks to display in other modules such as layout builder.

To achive this, the provided blocks will need to be updated on existing sites with the update hook provided, this will add the required node context to the CTA, releated links and releated topics blocks.
The existing $this->node method is still present for backward compatability for other blocks that extend servicesBlockBase until they are updated to use the node route context value.

andybroomfield committed:
Add layout builder support for service blocks

Ref #116

Add the node context to the service blocks, this allows them to be used
with layout builder

andybroomfield committed:
Update all services blocks to use node contexts

Set a node context to all service blocks and update the block base and all service
block functions to use the contextValue('node') if present.

Update each of the provided blocks to set a context_mapping for the current node route
Provide update hooks for CTA, releated links and related topics blocks
Provide a backward compatible way of accessing the current node via $this->node
for blocks that have not updated to use a node context so they can still get the current
node. (todo: decide if this should then be deprecated).

finnlewis commented:
Discussing in Merge Tuesday: 

 - @ekes thinks this makes sense insofar as we're removing the hard coding of a single node entity on the route, to allow configurable context (and support layout builder). 
 - @andybroomfield still want check / write update hooks to switch contexts - will review and take out of draft when ready to review.

andybroomfield committed:
Merge branch '2.x' into fix/116-layout-builder-block-support

andybroomfield committed:
Add layout builder support to the service landing template in the module

Check for layout builder support and if present render the layout builder instead
